app/forms/AssociateUcr.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import forms.AssociateKind.{Ducr, Mucr}
+import play.api.data.Forms._
+import play.api.data.format.{Formats, Formatter}
+import play.api.data.{Form, FormError, Forms}
+import play.api.libs.json.{Format, JsString, Json, JsonValidationError, Reads, Writes}
+import utils.validators.forms.FieldValidator._
+
+sealed abstract class AssociateKind(val formValue: String)
+
+object AssociateKind {
+  case object Mucr extends AssociateKind("mucr")
+  case object Ducr extends AssociateKind("ducr")
+
+  private val lookup = PartialFunction[String, AssociateKind] {
+    case Mucr.formValue => Mucr
+    case Ducr.formValue => Ducr
+  }
+
+  implicit val formatter: Formatter[AssociateKind] = new Formatter[AssociateKind] {
+    override def bind(key: String, data: Map[String, String]): Either[Seq[FormError], AssociateKind] = {
+      data.get(key).map { kind =>
+        lookup.andThen(Right.apply).applyOrElse(kind, (_: String) => Left(Seq(FormError(key, "error.unknown"))))
+      }
+    }.getOrElse(Left(Seq(FormError(key, "error.required"))))
+
+    override def unbind(key: String, value: AssociateKind): Map[String, String] = Map(key -> value.formValue)
+  }
+
+  implicit val format =
+    Format[AssociateKind](Reads.StringReads.collect(JsonValidationError("error.unknown"))(lookup), Writes(kind => JsString(kind.formValue)))
```

Comment:
What's wrong with Json.format?

Reply:
```
error] /home/lukasz.dubiel/hmrc/customs-movements-frontend/app/forms/AssociateUcr.scala:50: Sealed trait forms.AssociateKind is not supported: no known subclasses
[error]   implicit val format = Json.format[AssociateKind]
[error]                                    ^
```

Reply:
But now `AssociateKind` is a sealed trait and it should works (`Json.format`)

Reply:
I check it for now it does not.